consensus/tendermint/node_view.go

```diff
@@ -14,31 +14,42 @@ import (
 )
 
 type NodeView struct {
-	tmNode             *Node
-	validatorPublicKey crypto.PublicKey
-	txDecoder          txs.Decoder
-	runID              simpleuuid.UUID
+	tmNode    *Node
+	publicKey crypto.PublicKey
+	txDecoder txs.Decoder
+	runID     simpleuuid.UUID
 }
 
-func NewNodeView(tmNode *Node, txDecoder txs.Decoder, runID simpleuuid.UUID) (*NodeView, error) {
-	publicKey, err := crypto.PublicKeyFromTendermintPubKey(tmNode.PrivValidator().GetPubKey())
-	if err != nil {
-		return nil, err
+func NewNodeView(txDecoder txs.Decoder, runID simpleuuid.UUID) *NodeView {
+	return &NodeView{
+		txDecoder: txDecoder,
+		runID:     runID,
 	}
+}
+
+func (nv *NodeView) SetValidator(privVal types.PrivValidator) error {
+	pK, err := crypto.PublicKeyFromTendermintPubKey(privVal.GetPubKey())
+	nv.publicKey = pK
+	return err
+}
+
+func (nv *NodeView) SetNode(tmNode *Node) {
```

Comment:
Hmmm... I'm actually not sure `NodeView` should be instantiated at all without Tendermint. It is meant to be a wrapping of the Tendermint node. The only question is whether we have any use for a validator private key in this non-consensus mode. There are a couple of future places where we _might_ such as for adding the validator's signature to a mempool signed tx. But to be honest the single-node case is kind of out of scope now.

I think we should move this back to constructor.